testsuite/tests/singlecluster/gateway/reconciliation/change_targetref/test_update_authpolicy_target_ref.py

```diff
@@ -0,0 +1,42 @@
+"""
+Test for changing targetRef field in AuthPolicy
+"""
+
+import pytest
+
+from testsuite.kuadrant.policy.authorization.auth_policy import AuthPolicy
+
+pytestmark = [pytest.mark.kuadrant_only, pytest.mark.dnspolicy]
+
+
+@pytest.fixture(scope="module")
+def authorization(oidc_provider, gateway, cluster, blame, module_label, route):  # pylint: disable=unused-argument
+    """Overwrite the authorization fixture and attach it to the gateway"""
+    policy = AuthPolicy.create_instance(cluster, blame("authz"), gateway, labels={"testRun": module_label})
+    policy.identity.add_oidc("default", oidc_provider.well_known["issuer"])
+    return policy
+
+
+def test_update_auth_policy_target_ref(
+    gateway2, authorization, client, client2, auth, dns_policy, dns_policy2
+):  # pylint: disable=unused-argument
+    """Test updating the targetRef of an AuthPolicy from Gateway A to Gateway B"""
+    response = client.get("/get", auth=auth)
+    assert response.status_code == 200
+
+    response = client.get("/get")
+    assert response.status_code == 401
+
+    response = client2.get("/get")
+    assert response.status_code == 200
+
+    authorization.refresh().model.spec.targetRef = gateway2.reference
+    res = authorization.apply()
+    assert res.status() == 0, res.err()
+    authorization.wait_for_ready()
```

Comment:
You can put this into a fixture that returns a function you can call inside both of the tests. Also you can use `.modify_and_apply()` like this if you want, but no need to.
```py
@pytest.fixture(scope="session")
def change_target_ref():
    """Function that changes targetRef of given policy"""
    def _change_targetref(policy, gateway):
        def _apply_taget_ref(apiobj):
            apiobj.model.spec.targetRef = gateway.reference
            return True

        policy.modify_and_apply(_apply_taget_ref)
        assert policy.status() == 0, policy.err()
        policy.wait_for_ready()
    return _change_targetref

```